ext/TapirDynamicPPLExt.jl

```diff
@@ -0,0 +1,19 @@
+module TapirDynamicPPLExt
+
+if isdefined(Base, :get_extension)
+    using DynamicPPL: DynamicPPL
+    using Tapir: Tapir
+else
+    using ..DynamicPPL: DynamicPPL
+    using ..Tapir: Tapir
+end
+
+using Tapir: DefaultCtx, CoDual, NoPullback, primal, zero_fcodual
+
+# This is purely an optimisation.
+Tapir.@is_primitive DefaultCtx Tuple{typeof(DynamicPPL.istrans), Vararg}
```

Comment:
Good point. This should now be straightforward.

Reply:
Okay. It's not completely straightforward to create a macro to do this, because
1. the signatures for `@is_primitive` and `rrule!!`s are different from each other, meaning that some transformation between them is required, and
2. `Vararg`s at the end of the macro are very common, and they're a little tricky to handle in full generality.

This is to say that it would be fairly easy to create a very brittle macro to do this translation, but I suspect really quite hard (and time-consuming) to create a robust one. Given that this will only save 1 or 2 lines of code, and wouldn't be super safe in general (see the health warning associated to the new `simple_zero_adjoint` function), I don't think it's worth the hassle right now.

Instead, I've improved things a little bit by adding the function `simple_zero_adjoint`, which you can use to make use of the common pattern it encodes.